src/Form.php

```diff
@@ -0,0 +1,378 @@
+     * {@see getFirstErrors()}
+     */
+    public function getFirstError(string $attribute): string
+    {
+        return isset($this->errors[$attribute]) ? reset($this->errors[$attribute]) : '';
+    }
+
+    /**
+     * Returns the first error of every attribute in the model.
+     *
+     * @return array the first errors. The array keys are the attribute names, and the array values are the
+     * corresponding error messages. An empty array will be returned if there is no error.
+     *
+     * {@see getErrors()}
+     * {@see getFirstError()}
+     */
+    public function getFirstErrors(): array
+    {
+        if (empty($this->errors)) {
+            return [];
+        }
+
+        $errors = [];
+
+        foreach ($this->errors as $name => $es) {
+            if (!empty($es)) {
+                $errors[$name] = \reset($es);
+            }
+        }
+
+        return $errors;
+    }
+
+    /**
+     * Returns a value indicating whether there is any validation error.
+     *
+     * @param string|null $attribute attribute name. Use null to check all attributes.
+     *
+     * @return bool whether there is any error.
+     */
+    public function hasErrors(?string $attribute = null): bool
+    {
+        return $attribute === null ? !empty($this->errors) : isset($this->errors[$attribute]);
+    }
+
+    /**
+     * Populates the model with input data.
+     *
+     * which, with `load()` can be written as:
+     *
+     * ```php
+     * $body = $request->getParsedBody();
+     * $method = $request->getMethod();
+     *
+     * if ($method === Method::POST && $loginForm->load($body)) {
+     *     // handle success
+     * }
+     * ```
+     *
+     * `load()` gets the `'FormName'` from the model's [[formName()]] method (which you may override), unless the
+     * `$formName` parameter is given. If the form name is empty, `load()` populates the model with the whole of
+     * `$data`,
+     * instead of `$data['FormName']`.
+     *
+     * Note, that the data being populated is subject to the safety check by [[setAttributes()]].
+     *
+     * @param array $data the data array to load, typically `$_POST` or `$_GET`.
+     *
+     * @return bool whether `load()` found the expected form in `$data`.
+     */
+    public function load($data): bool
+    {
+        $result = false;
+
+        $scope = $this->getFormName();
+
+        if ($scope === null && !empty($data)) {
+            $this->setAttributes($data);
+            $result = true;
+        } elseif (isset($data[$scope])) {
+            $this->setAttributes($data[$scope]);
+            $result = true;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Sets the attribute values in a massive way.
+     *
+     * @param array $values attribute values (name => value) to be assigned to the model.
+
+     * {@see safeAttributes()}
+     * {@see attributes()}
+     */
+    public function setAttributes($values): void
+    {
+        if (\is_array($values)) {
```

Comment:
You should type the parameters `$values` as `array` instead

Reply:
done.